src/markets/edge/deploy.ts

```diff
@@ -363,115 +369,93 @@ export async function deployToEdge({
   extId,
   devChangelog = "",
   zip,
-  verbose: isVerbose
+  verbose,
+  dry = false
 }: EdgeOptions): Promise<boolean> {
-  return new Promise(async (resolve, reject) => {
-    const [width, height] = [1280, 720]
-    const puppeteerArgs =
-      process.env.NODE_ENV === "development"
-        ? {
-            headless: false,
-            defaultViewport: { width, height },
-            args: [`--window-size=${width},${height}`] //, "--window-position=0,0"]
-          }
-        : {}
-    const browser = await puppeteer.launch(puppeteerArgs)
-
-    const [page] = await browser.pages()
-    await disableImages(page)
-    await addLoginCookie({ page, cookie })
-    const urlStart = `${getBaseDashboardUrl(extId)}/packages/dashboard`
-
-    if (isVerbose) {
-      console.log(
-        getVerboseMessage({
-          market,
-          message: `Launched a Puppeteer session in ${urlStart}`
-        })
-      )
-    }
+  const [width, height] = [1280, 720]
+  const puppeteerArgs =
+    process.env.NODE_ENV === "test"
+      ? {
+          headless: false,
+          defaultViewport: { width, height },
+          args: [`--window-size=${width},${height} --window-position=0,0`]
+        }
+      : {}
+  const browser = await puppeteer.launch(puppeteerArgs)
+
+  const [page] = await browser.pages()
+  await disableImages(page)
+  await addLoginCookie({ page, cookie })
+  const urlStart = getDashboardUrl(extId)
+
+  const vLog = (message: string) =>
+    verbose &&
+    console.log(
+      getVerboseMessage({
+        market,
+        message
+      })
+    )
+
+  try {
+    vLog(`Launched a Puppeteer session in ${urlStart}`)
 
     await page.goto(urlStart)
 
-    try {
-      await openRelevantExtensionPage({ page, extId })
-    } catch (e) {
-      await browser.close()
-      reject(e)
-      return
-    }
+    await openRelevantExtensionPage({ page, extId })
 
-    if (isVerbose) {
-      console.log(
-        getVerboseMessage({
-          market,
-          message: "Opened relevant extension page"
-        })
-      )
-    }
+    vLog("Opened relevant extension page")
 
-    await cancelVersionInReviewIfNeeded({ page, isVerbose, zip })
+    await cancelVersionInReviewIfNeeded({ page, isVerbose: verbose, zip })
 
-    try {
-      if (await getIsInStore({ page })) {
-        await verifyNewVersionIsGreater({ page, zip })
-      }
-    } catch (e) {
-      await browser.close()
-      reject(e)
-      return
+    if (await getIsInStore({ page })) {
+      await verifyNewVersionIsGreater({ page, zip })
     }
 
     await uploadZip({ page, zip, extId })
 
-    if (isVerbose) {
-      console.log(
-        getVerboseMessage({
-          market,
-          message: `Uploading ZIP: ${zip}`
-        })
-      )
-    }
+    vLog(`Uploading ZIP: ${zip}`)
```

Comment:
This should be before the `await uploadZip` function. Otherwise, it should be `Uploaded ZIP`

Reply:
The uploading actually happens when it wait for the buttonNext to be clickable. Nontheless, I moved it up, might as well xD